### ISSUE TITLE
Cannot read property 'startsWith' of undefined

### ISSUE THREAD

**Issue by MichaelKuhl71** [asciidoctor/asciidoctor-vscode#509]
When I try to create a pdf from a adoc file out of VS Code I get an error message "Cannot read property 'startsWith' of undefined"

![image](https://user-images.githubusercontent.com/99040110/152573933-1fe33489-9600-4720-9dac-a92c47f46e63.png)

I have tested it with several asciidoc files. Html export or DocBook is working fine.

Sample file:

```adoc
= xxx
:revdate: 2202-02-04
:revnumber: 00.01
:revremark: Internal
:copyright: xxx
//:version-label!:
:author: xxx
:title: xxxx
:doctype: book
//:imagesdir: ./images
:toc: 
:toc-title: Tabel of contents
:toclevels: 3

:sectnums:
:sectnumlevels: 10
:xrefstyle: short

== Hello World

Test pdf

== Test

xxx Test pdf
```

VS Code:
Version: 1.63.2 (user setup)
Commit: 899d46d82c4c95423fb7e10e68eba52050e30ba3
Datum: 2021-12-15T09:40:02.816Z
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
Betriebssystem: Windows_NT x64 10.0.19043

AsciiDoc v2.9.3

**Comment by Mogztter** [comment]
Could you please open the Developer Tools and check if there's any error in the console? Also, could you please run the task (View > Command Palette): "Developer: Show logs" and check if you can find an error in "Extension Host"?

What I'm looking for is the complete stacktrace.

**Comment by joanna-grabowska** [comment]
Hi! I've got the same issue. Here's what my logs show:


```
[2022-02-05 16:49:38.414] [exthost] [error] TypeError: Cannot read property 'startsWith' of undefined
	at ExportAsPDF.<anonymous> (c:\Users\JG\.vscode\extensions\asciidoctor.asciidoctor-vscode-2.9.3\dist\src\commands\exportAsPDF.js:93:56)
	at Generator.next (<anonymous>)
	at fulfilled (c:\Users\JG\.vscode\extensions\asciidoctor.asciidoctor-vscode-2.9.3\dist\src\commands\exportAsPDF.js:5:58)
	at processTicksAndRejections (internal/process/task_queues.js:93:5) asciidoc.exportAsPDF {"value":"asciidoctor.asciidoctor-vscode","_lower":"asciidoctor.asciidoctor-vscode"}
```

**Comment by Mogztter** [comment]
Thanks, the related line is:

https://github.com/asciidoctor/asciidoctor-vscode/blob/1d96cb6d49365c07ed6f22fb8b6af70acdc5d6af/src/commands/exportAsPDF.ts#L95

I think this is a regression introduced in https://github.com/asciidoctor/asciidoctor-vscode/commit/cd477c734561e0b7c966fe62c9c193322dff9f7c where `!isNullOrUndefined(titlePageLogo)` was replaced by `!(titlePageLogo === null)`.

In this case, `titlePageLogo` is `undefined` and `undefined` is not strictly equals to `null`.
Instead we could use:

```js
if (titlePageLogo)
```

Which will return `true` if the value is not empty, null or undefined.

**Comment by ultrask14** [comment]
I also have this issue, it occures on multiple machines with a multide of files sizes. I dont believe the solution suggested could apply to myself. See example file content:

```
# test

test text
```

error logs below:
```
[2022-02-11 20:31:13.590] [exthost] [error] TypeError: Cannot read property 'startsWith' of undefined
	at ExportAsPDF.<anonymous> (c:\Users\tomok\.vscode\extensions\asciidoctor.asciidoctor-vscode-2.9.4\dist\src\commands\exportAsPDF.js:93:56)
	at Generator.next (<anonymous>)
	at fulfilled (c:\Users\tomok\.vscode\extensions\asciidoctor.asciidoctor-vscode-2.9.4\dist\src\commands\exportAsPDF.js:5:58)
	at processTicksAndRejections (internal/process/task_queues.js:93:5) asciidoc.exportAsPDF {"value":"asciidoctor.asciidoctor-vscode","_lower":"asciidoctor.asciidoctor-vscode"}
```

**Comment by Johan1974** [comment]
Hello,

I am getting the same and see the following line the error log

`[2022-02-17 12:44:10.832] [exthost] [error] TypeError: Cannot read property 'startsWith' of undefined
	at ExportAsPDF.<anonymous> (c:\Users\Johan\.vscode\extensions\asciidoctor.asciidoctor-vscode-2.9.4\dist\src\commands\exportAsPDF.js:93:56)
	at Generator.next (<anonymous>)
	at fulfilled (c:\Users\Johan\.vscode\extensions\asciidoctor.asciidoctor-vscode-2.9.4\dist\src\commands\exportAsPDF.js:5:58)
	at processTicksAndRejections (internal/process/task_queues.js:93:5) asciidoc.exportAsPDF {"value":"asciidoctor.asciidoctor-vscode","_lower":"asciidoctor.asciidoctor-vscode"}`

**Comment by Mogztter** [comment]
@Johan1974 Indeed, that's the same issue. It has been fixed https://github.com/asciidoctor/asciidoctor-vscode/commit/0f99b97e5a9e9a85e9d3a70ce1ba65001255ab11 and will be available in the next release.

**Comment by Johan1974** [comment]
My apologies for the double post. 

When wil this release be available? Can i go back to a certain release untill the new release is available? Tried the previous release but got the same error.

**Comment by MichaelKuhl71** [comment]
I did it with 2.8.10 , but unfortunately I had other problems with this old version (theme was not working, png was not working). Therefore I installed ruby and asciidoctor-pdf. With VS code I had to many problems,

**Comment by Mogztter** [comment]
> My apologies for the double post.

No worries.

> When wil this release be available? Can i go back to a certain release untill the new release is available? Tried the previous release but got the same error.

I just released a new version 2.9.5 which should fix this issue.

**Comment by Johan1974** [comment]
That is quick, let me try tomorrow. I will let you know the results.

Op do 17 feb. 2022 21:14 schreef Guillaume Grossetie <
***@***.***>:

**Comment by Johan1974** [comment]
Just checked all is good! 😁

Op do 17 feb. 2022 21:11 schreef Guillaume Grossetie <
***@***.***>:

**Comment by ultrask14** [comment]
> > My apologies for the double post.
> 
> No worries.
> 
> > When wil this release be available? Can i go back to a certain release untill the new release is available? Tried the previous release but got the same error.
> 
> I just released a new version 2.9.5 which should fix this issue.

I can also confirm its working. Thanks for the speedy fix!
